src/ts/generic/featurecollection.ts

```diff
@@ -192,10 +193,30 @@ function valueToType(value: boolean | number | string): ColumnType {
 }
 
 function introspectHeaderMeta(features: IFeature[]): HeaderMeta {
-    const feature = features[0];
-    const geometry = feature.getGeometry ? feature.getGeometry() : undefined;
-    const geometryType = geometry ? geometry.getType() : undefined;
-    const properties = feature.getProperties ? feature.getProperties() : {};
+    let geometryType: GeometryType = undefined;
+
+    for (const f of features) {
```

Comment:
1 thing I haven't really wrapped my head around is why there are 2 implementations of many of these functions, 1 under `generic/` and 1 under `geojson/`. I duplicated this logic in both places at the moment which obviously isn't ideal.

We could potentially pull it out into some helper, but I don't think the `IFeature` and `IGeoJsonFeature` types are interchangeable. So it might need to be something like `inferGeometryType(geomTypes: string[])`. Curious if anyone has thoughts or knows more of the backstory here

Reply:
I've tried to reduce the duplication in that generic contains most of the logic but as you note I've not been entirely successful. The two implementation targets geosjon and ol (OpenLayers) uses generic. IFeature is the generic interface and is inherited in IGeoJsonFeature and ISimpleFeature (OpenLayers) which are quite different in structure.

Reply:
Maybe something like this? https://github.com/flatgeobuf/flatgeobuf/pull/159/commits/60f952b140064135cd21015d5abb168138c57a29

It's a little clumsy b/c even tho `IGeoJsonFeature` extends `IFeature`, it lacks the `getGeometry` interface we would need to treat them consistently. I'm not sure if there's a more Typescipt-idiomatic way to do the type of disambiguation I'm doing in that function.

No rush on reviewing, I will probably be a way for a bit for Thanksgiving Holiday. :turkey:

Reply:
Had a look and I like what I see. :) Thanks.